classes/class-wc-connect-functions.php

```diff
@@ -83,5 +83,17 @@ public static function has_cart_or_checkout_block() {
 
 			return false;
 		}
+
+		/**
+		 * Checks whether the current user has permissions to manage shipping labels.
+		 *
+		 * @return boolean
+		 */
+		public static function user_can_manage_labels() {
+			/**
+			 * @since 1.25.14
+			 */
+			return apply_filters( 'wcship_user_can_manage_labels', current_user_can( 'manage_woocommerce' ) || current_user_can( 'wcship_manage_labels' ) );
```

Comment:
Curious on your thoughts regarding introducing a new permission versus re-using the existing `edit_shop_orders` (or combination of existing ones)? 

Is the `wcship_user_can_manage_labels` filter here to simply to give other plugins a chance to manipulate WCS label permission check? Should we allow other plugins to affect how WCS check permissions? On one side, it enhance compatibility; on the other side, it seems to open up label control.

Reply:
> Is the `wcship_user_can_manage_labels` filter here simply to give other plugins a chance to manipulate WCS label permission check?

That's correct, the intent is to make this flexible by either leveraging the new capability type _or_ by having a code snippet added by store owners (ideally) allow the change.

A lot of this discussion happened in this slack thread: p1623358576053300-slack-CKNC7KACA

We wouldn't want users with `edit_shop_orders` permission to print labels (since they could also change the plugin's configured packages via the shipping labels modal).
Another possibility is to have a wider refactor on the permission checks and the shipping modal to _not_ allow `edit_shop_orders` to change the plugin's configured packages 😬 

Let me know if you have other thoughts! :)

Reply:
Thanks for the slack link and I agree on the `edit_shop_orders` permission discussion. Unless we know exactly what permissions `edit_shop_orders` give, we should probably not use that.